### DISCUSSION TITLE
🎬 Video Recording Using Nintendo Switch Game Recording Feature

### DISCUSSION THREAD

**Discussion by Gr33nBug** [Dinones/Nintendo-Switch-Pokemon-Shiny-Hunter#65] [Ideas]
Right now the program has the option to record a videoclip of the encounters and save one when a shiny Pokémon is found.
This recording takes a toll in low spec machines either virtual or real hardware. As the fps gets lower, the program is more likely to desync the inputs and not recognize the images properly.

Some users might need to disable video recording in order to gain fps and use the program but may also want to have a videoclip of the shinies they have found. 

**I propose to add a new option to enable recording a video when a shiny Pokémon is found by pressing the Capture Button in the virtual joycon**

**Comment by Dinones** [comment]
Hi! @Gr33nBug!! Thank you for the idea! I didn't know you could use the button to record a video... I thought it could only be used to take screenshots. It'd be a nice implementation for low-performance devices, which, in fact, are more optimal than using a high-spec computer.

**Comment by Gr33nBug** [comment]
If you press the Capture Button for like 2 seconds the switch saves a video of the last 30 seconds of game session, enough for the shiny encounter.

I tought of give it a go and try to code it myself but I got a bit lost in the flow of the application...

**Reply by Dinones** [reply]
I like your idea, and I’ll add it to my to-do list!

Regarding the program flow, it's true it has become somewhat confusing _(to be honest, I didn’t anticipate the program would grow this much, or even that other people would help me)_. However, now that the program is stable, I plan to focus on rewriting and refactoring the code to make it easier to understand and more scalable before adding new features. In fact I've been working on it since a bit with the [State Machine diagrams](https://github.com/Dinones/Nintendo-Switch-Pokemon-Shiny-Hunter/tree/master/Media/Docs/State%20Machines).
